src/Symfony/Maker/MakeStateProvider.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Symfony\Maker;
+
+use Symfony\Component\Console\Exception\InvalidArgumentException;
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceNameCollectionFactoryInterface;
+use Symfony\Bundle\MakerBundle\ConsoleStyle;
+use Symfony\Bundle\MakerBundle\DependencyBuilder;
+use Symfony\Bundle\MakerBundle\Generator;
+use Symfony\Bundle\MakerBundle\InputConfiguration;
+use Symfony\Bundle\MakerBundle\Maker\AbstractMaker;
+use Symfony\Bundle\MakerBundle\Str;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+
+final class MakeStateProvider extends AbstractMaker
+{
+    private ResourceNameCollectionFactoryInterface $resourceNameCollection;
+
+    public function __construct(ResourceNameCollectionFactoryInterface $resourceNameCollection)
+    {
+        $this->resourceNameCollection = $resourceNameCollection;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getCommandName(): string
+    {
+        return 'make:state-provider';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getCommandDescription(): string
+    {
+        return 'Creates an API Platform state provider';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configureCommand(Command $command, InputConfiguration $inputConfig)
+    {
+        $command
+            ->addArgument('name', InputArgument::REQUIRED, 'Choose a class name for your state provider (e.g. <fg=yellow>AwesomeStateProvider</>)')
+            ->addArgument('resource-class', InputArgument::REQUIRED, 'Choose a Resource class')
+            ->setHelp(file_get_contents(__DIR__.'/Resources/help/MakeStateProvider.txt'));
+
+        $inputConfig->setArgumentAsNonInteractive('resource-class');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configureDependencies(DependencyBuilder $dependencies)
+    {
+    }
+
+    public function interact(InputInterface $input, ConsoleStyle $io, Command $command)
```

Comment:
`interact()` methods in both makers are the same, I may extract it into an abstract class or an external service

Reply:
I suggest using a trait marked `@internal`.